tests/models/test_parameterized_query.py

```diff
@@ -166,7 +166,8 @@ def test_validates_enum_list_value_parameters(self):
         "redash.models.parameterized_query.dropdown_values",
         return_value=[{"value": "1"}],
     )
-    def test_validation_accepts_integer_values_for_dropdowns(self, _):
+    @patch("redash.models.parameterized_query.query_has_parameters", return_value=False)
```

Comment:
I'm not too happy about stubbing out implementation details such as `query_has_parameters`. Is there a way to setup the data to organically provide this behavior?

Reply:
Unfortunately my experience with the backend unit tests is very limited :sweat_smile:, for this one I just cloned the behavior it had for `load_results` since it was returning a "There is no app context". 

Another option I have in mind is getting rid of that method and creating a `get_query_by_id_and_org` (so the stub would be a query, not a boolean)

Reply:
A few months later and I'm a little bit better at this now 🙂. So, turns out the `test_parameterized_query` is based in `TestCase`, thus it doesn't have an App Context, we then can:
1. Make it based in `BaseTestCase`, so that it will have an App Context for us to create an actual query. Downside: will make it slow;
2. Keep it like that, add a `_load_dropdown_query` that we can mock and keep mocking `dropdown_values` and a fast running test;

Any preference in this case?

Reply:
This test was actually harder to set with the organic App Context than I thought, so I'm back with the mocks, but I've updated the `query_has_parameters` to be a mock on the Query object.